packages/js/browserinstance.js

```diff
@@ -725,8 +725,12 @@ class BrowserInstance {
         try {
             await this.driver.wait(async () => {
                 obj = await this.executeScript(function(titleOrUrl) {
+                    let isMatched = document.title.toLowerCase().indexOf(titleOrUrl.toLowerCase()) != -1 || window.location.href.indexOf(titleOrUrl) != -1;
+                    let documentMatch = document.title.match(titleOrUrl);
+                    let locationMatch = window.location.href.match(titleOrUrl);
+                    isMatched = isMatched || documentMatch && documentMatch[0] === document.title || locationMatch && locationMatch[0] === window.location.href;
```

Comment:
I'd put parenthesis around `documentMatch && documentMatch[0] === document.title` and `locationMatch && locationMatch[0] === window.location.href`. 

Also, maybe rename `documentMatch` to `titleRegexMatch` and `locationMatch` to `urlRegexMatch` or something like that.

Reply:
Done !